integration-test-groups/aws2/aws2-s3/src/main/java/org/apache/camel/quarkus/component/aws2/Aws2S3Resource.java

```diff
@@ -97,18 +123,152 @@ public Response read(@PathParam("key") String key) throws Exception {
     @Path("s3/object-keys")
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<String> objectKey() throws Exception {
+    public List<String> objectKey(@QueryParam("bucket") String bucket) throws Exception {
+        if (bucket == null) {
+            bucket = bucketName;
+        }
+
         final List<S3Object> objects = (List<S3Object>) producerTemplate.requestBody(
-                componentUri(AWS2S3Operations.listObjects),
+                componentUri(bucket, AWS2S3Operations.listObjects) + "&autoCreateBucket=true",
                 null,
                 List.class);
         return objects.stream().map(S3Object::key).collect(Collectors.toList());
     }
 
-    private String componentUri(final AWS2S3Operations operation) {
+    @Path("s3/upload")
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    public String multipartUpload(@MultipartForm MultipartFormDataInput input) throws Exception {
+
+        Map<String, List<InputPart>> formDataMap = input.getFormDataMap();
+        String fileName = formDataMap.get("fileName").get(0).getBodyAsString();
+
+        if (fileName == null) {
+            throw new FileNotFoundException();
+        }
+
+        File file = File.createTempFile("aws2-", ".tmp");
+
+        List<InputPart> inputParts = formDataMap.get("file");
+        for (InputPart inputPart : inputParts) {
+            InputStream inputStream = inputPart.getBody(InputStream.class, null);
+            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
So we are upload only the second part ?

Reply:
This test is migrated from the camel one ```S3MultipartUploadOperationIT.java``` which runs a test to upload a file. I just check the S3 MultipartUpload operation in camel-aws2-s3 component. It looks like that it splits a file content into pieces by partSize. Then upload them in S3 in one request. I think this mulitpart upload does not  like the HTTP_MULTIPART_FORM. Anyway, I'm not an expert of AWS S3 so it mghit be some misunderstanding here.

Reply:
Not an expert too, trying to sort it out :) From the readings, please find a proposal below:
- The test send a simple string content and a key to this method (so we don't use @ MultipartForm)
- The string content is put in a `aws2-*.tmp` temp file
- We send this temp file and key with `multiPartUpload=true`
- Optionally, we try to make AWS2 library to chunk the content with low partSize for instance `partSize=1`

Would such a test makes sense ?

Reply:
Thanks @aldettinger - it makes sense.

Reply:
It looks like we can set ```partSize=1```` which throws
```
Caused by: software.amazon.awssdk.services.s3.model.S3Exception: Your proposed upload is smaller than the minimum allowed object size.
```

Reply:
Acoording to https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html
```
Each part must be at least 5 MB in size, except the last part. There is no size limit on the last part of your multipart upload.
```

I think we should add this limit in the camel-aws2-s3 document.

Reply:
And ```getObject``` after uploading, there is a ```chunk-signature``` fragment in the response result. I'm not sure that it could be handled by camel-aws2-s3 component or by users themself ? In current upload test, I just strip it.

Reply:
Concerning this PR, I think uploading a small file without partSize should be enough. We cover createMultipartUploadRequest, createMultipartUpload, uploadPart and completeMultipartUpload.

Concerning follow up issues to improve camel-aw2-s3 component. Checking partSize >= 5MB is a good idea, ideally the 5MB value should come from amazon in case it changes. Concerning getObject/chunk-signature, I have no idea. Maybe opening camel ticket would allow to collect feedback.